ecleankernel/process.py

```diff
@@ -120,9 +101,22 @@ def unprefixify(filenames: typing.Iterable[str]
             candidates = kernels
 
         for k in candidates:
+            def get_kernel_images(kernel_files: typing.List[GenericFile]
+                                 ) -> typing.Iterable[KernelImage]:
+                return (kernel_file.path
+                        for kernel_file in kernel_files
+                        if type(kernel_file) is KernelImage)
```

Comment:
I don't think you need to do this, you can just match against all files.

Now if you wanted to optimize this, you could collect `(st_dev, st_ino)` tuples for files and match against that but I don't think it's technically necessary here.

Reply:
Removed. I just thought it would be silly (though not strictly wrong) to match against all files. In `lilo.conf` (and I believe `grub.cfg`) I do only expect to find a reference to a `KernelImage`.

Reply:
I know but I think it's better to keep the code simple.